internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -390,6 +423,35 @@ func (r *AgentServiceConfigReconciler) newAgentRoute(instance *adiiov1alpha1.Age
 	return route, mutateFn
 }
 
+func (r *AgentServiceConfigReconciler) newAgentLocalAuthSecret(instance *adiiov1alpha1.AgentServiceConfig) (*corev1.Secret, controllerutil.MutateFn, error) {
+	publicKey, privateKey, err := gencrypto.ECDSAKeyPairPEM()
```

Comment:
Generating key pairs is typically compute-intensive relative to other activities. I think it would be better to create an empty Secret, then in the mutate function check whether the values have been set, and if not, then generate the key pair and set the values.